src/ground_script_1.c

```diff
@@ -0,0 +1,1863 @@
+#include "global.h"
+#include "debug.h"
+#include "ground_script.h"
+#include "ground_map.h"
+#include "dungeon.h"
+
+#ifndef NONMATCHING
+#define GROUND_SCRIPT_INCOMPLETE_DECLARATIONS
+#endif
+
+#include "ground_main.h"
+#include "code_80A26CC.h"
+
+typedef struct {
+    u8 xTiles;
+    u8 xFlags;
+    u8 yTiles;
+    u8 yFlags;
+} PosInfo;
+typedef struct GroundObjectData {
+    u8 kind;
+    u8 unk1;
+    u8 widthTiles;
+    u8 heightTiles;
+    PosInfo pos;
+    ScriptCommand *scripts[4]; // 2 - dialogue script
+} GroundObjectData;
+typedef struct GroundEffectData {
+    u8 kind;
+    u8 unk1;
+    u8 widthTiles; //???
+    u8 heightTiles;
+    PosInfo pos;
+    ScriptCommand *script;
+} GroundEffectData;
+
+void GroundSprite_Reset(s16);
+void GroundMap_Select(s16);
+void GroundMap_SelectDungeon(s16, DungeonLocation*, u8);
+void GroundMap_ExecuteEnter(s16);
+void GroundMap_ExecuteEvent(s16, bool8);
+void GroundMap_GetStationScript(ScriptInfoSmall *out, s16, s32, s32);
+void GroundMap_ExecuteStation(s32, s32, s32, bool8);
+void GroundLives_ExecuteScript(s32, s16 *, ScriptInfoSmall *);
+void GroundObject_ExecuteScript(s32, s16 *, ScriptInfoSmall *);
+void GroundEffect_ExecuteScript(s32, s16 *, ScriptInfoSmall *);
+void GroundLink_Select(s16);
+void GroundLives_Select(s16, s32 group, s32 sector);
+void GroundObject_Select(s16, s32 group, s32 sector);
+void GroundEffect_Select(s16, s32 group, s32 sector);
+void GroundEvent_Select(s16, s32 group, s32 sector);
+void GroundLives_Cancel(s32 group, s32 sector);
+void GroundObject_Cancel(s32 group, s32 sector);
+void GroundEffect_Cancel(s32 group, s32 sector);
+void GroundEvent_Cancel(s32 group, s32 sector);
+void GroundLives_CancelBlank_1(void);
+void GroundObject_CancelBlank(void);
+void GroundEffect_CancelBlank(void);
+void GroundWeather_Select(s16);
+u32 GroundLives_ExecutePlayerScriptActionLives();
+s16 GroundObject_Add(s16 id, GroundObjectData*, s16 group, s8 sector);
+s16 GroundEffect_Add(s16 id, GroundEffectData*, s16 group, s8 sector);
+u32 FlagCalc();
+u32 FlagJudge();
+s32 OtherRandInt(s32); // IT RETURNS s32!! Not s16!
```

Comment:
That's the current definition in `other_random.h` so should be good to use that instead.